feature-libs/customer-ticketing/components/customer-ticketing-dialog/customer-ticketing-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import {
+  CustomerTicketingConfig,
+  MAX_ENTRIES_FOR_ATTACHMENT,
+  MAX_INPUT_CHARACTERS,
+  MAX_INPUT_CHARACTERS_FOR_SUBJECT,
+  MAX_SIZE_FOR_ATTACHMENT,
+} from '@spartacus/customer-ticketing/root';
+import {
+  FilesFormValidators,
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+
+@Directive()
+export abstract class CustomerTicketingDialogComponent {
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+  inputCharactersLimit: number = this.getInputCharactersLimit;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  get messagesCharacterLeft(): number {
+    return (
+      this.inputCharactersLimit - (this.form.get('message')?.value?.length || 0)
+    );
+  }
+
+  get allowedTypes(): string[] | undefined {
+    return this.customerTicketingConfig.customerTicketing
+      ?.attachmentRestrictions?.allowedTypes;
```

Comment:
should we have a default constant and value for allowed types as well?

Reply:
If the value is null or empty array, it will allow all types of files to be uploaded.
So, I think it's better if we don't defined the default constant.